src/installer/records.py

```diff
@@ -36,50 +50,77 @@ def parse(cls, h):
+
         self.path = path
         self.hash_ = hash_
         self.size = size
 
     def __repr__(self):
         # type: () -> str
-        return "RecordItem(path={!r}, hash_={!r}, size={!r})".format(
+        return "Record(path={!r}, hash_={!r}, size={!r})".format(
             self.path, self.hash_, self.size,
         )
 
     @classmethod
-    def parse(cls, path, hash_, size):
-        # type: (str, str, str) -> RecordItem
-        """Build a Record from parsing elements in a record row.
+    def from_elements(cls, path, hash_, size):
+        # type: (FSPath, str, str) -> Record
+        """Build a Record object, from values of the elements.
+
+        All arguments must be in string form, except `path` which can also be a
+        ``pathlib.Path`` instance on Python 3.
 
         Typical usage::
 
             reader = csv.reader(f)
             for row in reader:
-                record = Record.parse(row[0], row[1], row[2])
+                record = Record.from_elements(row[0], row[1], row[2])
 
-        All arguments are in string form. Meaning of elements are specified in
-        PEP 376. Raises ``ValueError`` if any of the elements is invalid.
+        All arguments are in string form. Meaning of each element is specified in
+        PEP 376. Raises ``InvalidRecord`` if any element is invalid.
         """
-        return cls(
-            path=path,
-            hash_=Hash.parse(hash_) if hash_ else None,
-            size=int(size) if size else None,
-        )
-
-
-def parse_record_file(f):
-    # type: (Iterator[str]) -> Iterator[RecordItem]
-    for row_index, row in enumerate(csv.reader(f)):
-        if len(row) > 3:
-            warnings.warn(
-                "Dropping columns [3:] from row {}".format(row_index),
-                SuperfluousRecordColumnsWarning,
+        # Validate the passed values.
+        issues = []
+
+        if not path:
+            issues.append("`path` cannot be empty")
+
+        if hash_:
+            try:
+                hash_value = Hash.parse(hash_)  # type: Optional[Hash]
+            except ValueError:
+                issues.append("`hash` does not follow the required format")
+        else:
+            hash_value = None
+
+        if size:
+            try:
+                size_value = int(size)  # type: Optional[int]
+            except ValueError:
+                issues.append("`size` cannot be non-integer")
+        else:
+            size_value = None
+
+        if issues:
+            raise InvalidRecord(elements=(path, hash_, size), issues=issues)
+
+        return cls(path=path, hash_=hash_value, size=size_value,)
+
+
+def parse_record_file(rows):
+    # type: (Iterator[str]) -> Iterator[Record]
+    """Parse a RECORD file, provided as an iterator of record lines.
+    """
+    reader = csv.reader(rows, delimiter=",", quotechar='"', lineterminator=os.linesep)
+    for row_index, elements in enumerate(reader):
+        if len(elements) != 3:
```

Comment:
I wonder if forward-compatibility should be an consideration here, i.e. if the wheel spec add more columns to the record, the installer should be tolerant to the optional values. Hard-error on less than three columns is perfectly fine, of course.

Reply:
Maybe? I think it's better to be strict on what we have today, instead of trying to anticipate how we might change things in the future and accommodating for them now.

I'm pretty sure installer will need to be updated for some reason or another, when a new wheel format version is created. #16 has a Wheel-Version check embedded, so if we don't change that, then there's a clear definition of which versions of the wheel format are supported in a certain version of this library.

Support for newer Wheel format versions needs a newer versions of this library - that's a very reasonable approach to compatibility with newer wheel versions IMO.

Reply:
OK then.

Reply:
@uranusjr That might be since you don't have the commit/triage bit on this?